diagram-viewer/src/main/java/com/powsybl/diagram/viewer/sld/SingleLineDiagramModel.java

```diff
@@ -32,6 +30,21 @@
  */
 public class SingleLineDiagramModel extends DiagramModel {
 
+    public enum VoltageLevelLayoutFactoryType {
+        POSITION_WITH_EXTENSIONS, POSITION_BY_CLUSTERING, CGMES, RANDOM, SMART;
+
+        @Override
+        public String toString() {
+            return switch (this) {
+                case SMART -> "Smart";
```

Comment:
Maybe `"'Smart' choice"`, to highlight the choice behind? because the SmartSelector chooses position with extensions if the vl contains at least one position extension, if not it chooses cgmes diagram layout if there's at least one cgmes extension, if not it chooses clustering.